packages/driver/src/cypress/screenshot.coffee

```diff
@@ -94,6 +128,15 @@ validate = (props, cmd, log) ->
 
     values.clip = clip
 
+  if padding = props.padding
+    if not (_.isFinite(padding) or (_.isArray(padding) and padding.length >=1 && padding.length <= 4 and _.every(padding, _.isFinite)))
```

Comment:
abstract these methods into human readable forms because the logic density here is too high to easily deconstruct what the rules are...

Reply:
I've left this one as is as I'm not too sure how to make this one more readable, but I'm happy to apply a suggestion if you can show how you want it to look.

Reply:
I've reformatted this. But I'm happy to apply an alternative suggestion if you can show how you want it to look.

Reply:
I think Brian is suggesting pulling the logic into a function or variable so that it has some meaning assigned to it and isn't just a bunch of boolean logic strung together. Something like:

```coffeescript
  isNumberOrArrayOfNumbers = (padding) ->
    return (
      _.isFinite(padding) or
      (_.isArray(padding) and
        padding.length >= 1 and
        padding.length <= 4 and
        _.every(padding, _.isFinite))
    )

  if padding = props.padding
    if not isNumberOrArrayOfNumbers(padding)
      ## ...
```

Reply:
@NMinhNguyen I suggested via DM something like `not (isFiniteNumber or isPaddingShorthand)`.

Reply:
The thing is that if you look just above at `clip`, there's arguably even more validation logic in a single `if`, but I'm happy to take @chrisbreiding's suggestion.